hardware_interface/CMakeLists.txt

```
@@ -73,6 +73,9 @@ if(BUILD_TESTING)
 
   find_package(ament_cmake_gmock REQUIRED)
   find_package(ros2_control_test_assets REQUIRED)
+  find_package(ament_index_cpp REQUIRED)
+  find_package(PkgConfig REQUIRED)
+  pkg_check_modules(LIBXML2 REQUIRED IMPORTED_TARGET libxml-2.0)
```

Why not just this? https://cmake.org/cmake/help/latest/module/FindLibXml2.html
```suggestion
  find_package(LibXml2)
```